src/CBLTLSIdentity_Internal.hh

```diff
@@ -328,6 +325,11 @@ public:
         else       return new CBLTLSIdentity(nullptr, new CBLCert(cert.get()));
     }
     
+    static CBLTLSIdentity* _cbl_nullable IdentityWithCerts(CBLCert* cert) {
+        std::scoped_lock<std::mutex> lock(_mutex);
```

Comment:
Probably don't mutex, because we merely pass the input to a new CBLTLSIdentity object.

Reply:
I updated the code in that method to also verify the private key associated with the given certificates. Can you check if I should have the mutex? I am not 100% sure when to have the mutex lock.

Reply:
I think we don't need it because we merely put the input into a newly created CBLKeyPair.